Collected small changes and fixes

akohlmey opened the pull request:
**Summary**

This pull request combines multiple small changes and fixes.

**Related Issue(s)**

Fixes #4797 

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Artificial Intelligence (AI) Tools Usage**

The GitHub Copilot coding instructions were mostly updated by GitHub Copilot.

**Backward Compatibility**

N/A

**Implementation Notes**

The following individual changes are included:
- remove references to https://ci.lammps.org and update information on automated testing
- update GitHub checkout action for testing workflows to version 6
- update GitHub coding instructions to follow best practices
- include LAMMPS version date only for 'release' and 'stable' branch when creating the manual. otherwise use 'Maintenance' (for 'maintenance 'branch) or 'Development' (for everything else)
- add deprecation messages for recently removed styles
- add more doxygen comments for LAMMPS class and Pointers class
- some more improvements for binary packaging on Linux
- remove some debug code
- reduce compiler warnings
- improve error messages and discussion of error details
- fix logic bug from last minute change in PR #4772
- update metadata in mW potential file in INTEL package tests

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system

akohlmey committed:
update checkout action in github workflows to version 6

akohlmey committed:
improve packaging and wrapper script for Linux some more

akohlmey committed:
reorder initializer and remove debug output

akohlmey committed:
fix permission of LAMMPS shared library

akohlmey committed:
remove references to https://ci.lammps.org/ and update description of automated tests

akohlmey committed:
Merge branch 'develop' into collected-small-changes

stanmoore1 committed:
Kokkos_ENABLE_HIP_MULTIPLE_KERNEL_INSTANTIATIONS is already set in KOKKOS.cmake

stanmoore1 committed:
Add CMake preset file for elcap

stanmoore1 committed:
Tweak library link line

akohlmey committed:
we don't use the sphinx toolbox extension. drop it.

akohlmey committed:
improve choice of version string displayed in manual

akohlmey committed:
correct description

akohlmey committed:
add explanation for deprecation error messages

akohlmey committed:
fix atc, fix poems, and fix nve/awpmd are no longer available

akohlmey committed:
improve error messages

Copilot committed:
Improve Copilot coding agent instructions with CI debugging, documentation, and sample prompts

Co-authored-by: akohlmey <289251+akohlmey@users.noreply.github.com>

akohlmey committed:
rename LAMMPS::match_style() function to more obvious LAMMPS::get_style_pkg()

akohlmey committed:
improve and expand explanation for "Unrecognized style" errors

akohlmey committed:
improve error messages

akohlmey committed:
update review instructions to search for alternative tokens for logical operations

akohlmey committed:
small corrections for tests and consistency

akohlmey committed:
improve embedded documentation for LAMMPS and Pointers classes, add more info

akohlmey committed:
improved clear command text and format

akohlmey committed:
add Error::COMMAND constant and document it in the developer notes

akohlmey committed:
more style deprecation updates

akohlmey committed:
add link to Index to navigation bar

stanmoore1 committed:
Fix logic bug

akohlmey committed:
fore error message improvements in input processing

akohlmey committed:
update death tests for echo command

akohlmey committed:
improve error messages

akohlmey committed:
some more error message improvements

stanmoore1 commented on src/lammps.cpp:
```diff
@@ -965,10 +996,13 @@ void LAMMPS::post_create()
-   initialize top-level classes
-   do not initialize Timer class, other classes like Run() do that explicitly
-------------------------------------------------------------------------- */
+/** Initialize the LAMMPS top-level classes
```
I think this type of comment style is harder to read and doesn't match the rest of LAMMPS.

akohlmey replied:
Show me a different way to transfer that kind of information into the manual without redundancies.  Some things take some getting used to, but the payoff in the long run is worth it.

We've transformed the C-language and Python-language interface so have only one documentation and set of comments so it is easier to keep it all consistent and up-to-date. The C++ programming interface is woefully missing; mostly because it is a lot of work. Since I found that GitHub Copilot is quite good at producing a usable draft for those comments, I am now working to have at least the generally usable members of the base classes documented.

stanmoore1 replied:
I think we need to get input from other developers on this, e.g. @athomps and/or @sjplimp.

akohlmey replied:
> I think we need to get input from other developers on this, e.g. @athomps and/or @sjplimp.

I understand the desire to keep things the way they are, but we are already too far down the road of adding Doxygen comments to reverse this process. The benefits, especially in the long run, outweigh the discomfort, IMO.  Like with other "modernization" changes (even though Doxygen is hardly modern), it takes some time to get used to it, embrace it, and use it to the advantage of LAMMPS users and developers, especially future ones.  Happy to continue discussing this in private.

akohlmey replied:
> I think we need to get input from other developers on this, e.g. @athomps and/or @sjplimp.

@stanmoore1 FYI, I've moved the doxygen related changes out of this pull request into #4793 to have them all in one place and not block merging this PR due to this discussion.

stanmoore1 replied:
Thank you @akohlmey

stanmoore1 commented on src/pointers.h:
```diff
@@ -98,41 +100,40 @@ class Pointers {
-  class AtomKokkos *&atomKK;
-  class MemoryKokkos *&memoryKK;
-  class Python *&python;
+  LAMMPS *lmp;            ///< LAMMPS instance
```
I think `///<` is ugly.

akohlmey replied:
Please check out https://doxygen.nl/manual/docblocks.html#memberdoc for alternatives. Happy to apply a different one.

stanmoore1 commented on src/lammps.cpp:
```diff
@@ -891,14 +913,23 @@ void LAMMPS::create()
-------------------------------------------------------------------------- */
+/** Perform LAMMPS class tasks that require all base classes being set up
+ *
+\verbatim embed:rst
```
Why are there `rst` commands in the comment? I think this makes the comment block harder to read.

akohlmey replied:
This is the only way to insert references to other parts of the manual and write embedded documentation that is consistent with the rest of the manual.

While looking at the docs for the latest version of breathe, I've noticed an alternative tag where the asterisks can remain. Need to check how recent this is and if it works for all existing cases that we already have. https://breathe.readthedocs.io/en/latest/markups.html#handling-leading-asterisks

stanmoore1 reviewed (commented):
Most of the changes look good but I'm concerned that adding doxygen and rst commands into comment blocks makes them ugly, harder to read, and doesn't match the rest of LAMMPS.

akohlmey committed:
make more compact and consistent

akohlmey committed:
rename CMAKE_CUSTOM_LINKER CMake variable to LAMMPS_CUSTOM_LINKER

For clarity and to avoid clashes, we should not define or use any
custom variables that have a name starting with CMAKE_

akohlmey committed:
update discussions of testing / code analysis link to software engineering paper

akohlmey committed:
move doxygen related changes out of this branch to PR #4793

akohlmey committed:
use consistent include guards

akohlmey committed:
correct metadata for mW potential file
